core/cmd/local_client.go

```diff
@@ -156,15 +156,15 @@ func initLocalSubCmds(client *Client, safe bool) []cli.Command {
 			Name:        "db",
 			Usage:       "Commands for managing the database.",
 			Description: "Potentially destructive commands for managing the database.",
-			Before: func(ctx *clipkg.Context) error {
-				return client.configExitErr(client.Config.ValidateDB)
-			},
 			Subcommands: []cli.Command{
 				{
 					Name:   "reset",
 					Usage:  "Drop, create and migrate database. Useful for setting up the database in order to run tests or resetting the dev database. WARNING: This will ERASE ALL DATA for the specified database, referred to by CL_DATABASE_URL env variable or by the Database.URL field in a secrets TOML config.",
 					Hidden: safe,
 					Action: client.ResetDatabase,
+					Before: func(ctx *clipkg.Context) error {
+						return client.configExitErr(client.Config.ValidateDB)
+					},
```

Comment:
Maybe consider defining a helper?
```suggestion
					Before: client.ValidateDBConfig,
```

```go
func (c *Client) ValidateDBConfig(ctx *clipkg.Context) error {
	return c.configExitErr(c.Config.ValidateDB)
},
```